packages/app/src/Pages/settings/Profile.tsx

```diff
@@ -112,6 +146,67 @@ export default function ProfileSettings(props: ProfileSettingsProps) {
     }
   }
 
+  async function onNip05Change(e: React.ChangeEvent<HTMLInputElement>){
+    const Nip05Address = e.target.value;
+    setNip05(Nip05Address)
+    const Nip05AddressElements = Nip05Address?.split('@') ?? [];
+    if (Nip05Address.length === 0) {
+        setNip05AddressValid(false)
+        setInvalidNip05AddressMessage("")
+    }else if(Nip05AddressElements.length < 2){
+        setNip05AddressValid(false)
+        setInvalidNip05AddressMessage(formatMessage(messages.InvalidNip05Address))
+    }
+    else if(Nip05AddressElements.length === 2){
+      nip05NostrAddressVerification(Nip05AddressElements.pop(), Nip05AddressElements.pop());
+    }
+    else{
+      setNip05AddressValid(false)
+    }
+  }
+
+  async function onLimitCheck(val:string, field:string){
+    if(field === "username"){
+      setName(val);
+      if(val?.length >= MaxUsernameLength){
+        setUsernameValid(false)
+        setInvalidUsernameMessage(formatMessage(messages.UserNameLengthError))
+      }else{
+        setUsernameValid(true)
+        setInvalidUsernameMessage("")
+      }
+    }
+    else if(field === "about"){
+      setAbout(val);
+      if(val?.length >= MaxAboutLength){
+        setAboutValid(false)
+        setInvalidAboutMessage(formatMessage(messages.AboutLengthError))
+      }else{
+        setAboutValid(true)
+        setInvalidAboutMessage("")
+      }
+    }
+  }
+
+  async function nip05NostrAddressVerification(nip05Domain: string | undefined, nip05Name: string| undefined) {
```

Comment:
This should also use `useEffect` with the `debounce` otherwise its going to make a request for every char that you enter or delete, you will also have a problem when 1 request overwrites the result if its slower/faster.